velox/connectors/hive/storage_adapters/gcs/GCSFileSystem.cpp

```diff
@@ -312,13 +325,31 @@ class GCSFileSystem::Impl {
           << "Config hive.gcs.json-key-file-path is empty or key file path not found";
     }
 
+    if (accessTokenEnabled == "true") {
+      std::string providerClassName =
+          hiveConfig_->config()->get<std::string>("gcs.access_token_provider", "");
+        throw std::runtime_error("AccessTokenProvider not configured. Please implement an out of band access token provider.");
```

Comment:
Should we add some checks here?
I would LOG an error here and let the GCS library handle the invalid config